independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java

```diff
@@ -356,6 +358,86 @@ protected abstract METHOD createResourceMethod(MethodInfo info, ClassInfo actual
     protected List<ResourceMethod> createEndpoints(ClassInfo currentClassInfo,
             ClassInfo actualEndpointInfo, Set<String> seenMethods, Set<String> existingClassNameBindings,
             Set<String> pathParameters, String resourceClassPath, boolean considerApplication) {
+
+        List<ResourceMethod> ret = new ArrayList<>();
+
+        List<FoundEndpoint> endpoints = collectEndpoints(currentClassInfo, actualEndpointInfo, seenMethods,
+                existingClassNameBindings, considerApplication, httpAnnotationToMethod, index, applicationScanningResult,
+                getAnnotationStore());
+        Map<String, BasicResourceClassInfo> basicResourceClassInfoMap = new HashMap<>();
+        for (FoundEndpoint endpoint : endpoints) {
```

Comment:
I can't really understand the changes here. Can you elaborate?

Reply:
`createEndpoints` does 2 things:

- determine if resource method is an endpoint
- create endpoint

I had a PR back in the days where I needed to collect all endpoints and you pointed out I should keep the logic in one place (one version of https://github.com/quarkusio/quarkus/pull/26567). The changes here just try to divide the two things without adding extra action. So for performance reasons, with each endpoint method it remembers its class, classNameBindings and httpMethod. If you look couple lines below, the logic is same -> lines are just moved to `collectEndpoints`.